mod_ood_proxy/lib/ood/nginx_stage.lua

```diff
@@ -4,21 +4,23 @@
 
   Start PUN process for given user
 --]]
-function pun(r, bin, user, app_init_url, exports, pre_hook_root_cmd)
+function pun(r, bin, user, app_init_url, exports, pre_hook_root_cmd, rails_config_hosts)
   local cmd = bin .. " pun -u '" .. r:escape(user) .. "'"
   local err
+  local env_table = {}
 
   if app_init_url then
     cmd = cmd .. " -a '" .. r:escape(app_init_url) .. "'"
   end
 
   if pre_hook_root_cmd then
-    local env_table = exports_to_table(r, exports)
+    env_table = exports_to_table(r, exports)
```

Comment:
This means we're forking every time now which could be costly. I was going to suggest to just set the environment variable directly, but it would need to be a part of the exports either way.

I wonder how expensive this fork it is relative to the other code path.

Reply:
`capture2e_with_env` forks within this lua code where `capture2e` boots the PUN within the current thread.

https://github.com/OSC/ondemand/blob/b7e60d5fecfd47b38ac7fc7e65a2a9095680b40c/mod_ood_proxy/lib/ood/nginx_stage.lua#L104

Reply:
Ah good point. I changed it to old behavior and just modify non-forked environment if not using pre pun hook.  I verified this still works using container.